src/EnergyPlus/PluginManager.cc

```diff
@@ -1459,6 +1469,22 @@ void PluginManager::runSingleUserDefinedPlugin([[maybe_unused]] EnergyPlusData &
 }
 #endif
 
+int PluginManager::getUserDefinedCallbackIndex(EnergyPlusData &state, const std::string &callbackProgramName)
+{
+    for (int i = 0; i < state.dataPluginManager->userDefinedCallbackNames.size(); i++) {
+        if (state.dataPluginManager->userDefinedCallbackNames[i] == callbackProgramName) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+void PluginManager::runSingleUserDefinedCallback(EnergyPlusData &state, int index)
+{
+    if (state.dataGlobal->KickOffSimulation) return;                      // Maybe?
+    state.dataPluginManager->userDefinedCallbacks[index]((void *)&state); // Check Index first
+}
```

Comment:
User defined components can execute the callback by index here.  I'm not sure if checking the index would be excessive here or not.  This isn't something a user can manipulate, either the user defined component gets a valid index at GetInput, or it doesn't and fails.  So I think this is fine.  I'm also open to suggestion about the kickoff simulation check.  Again, it seems quite reasonable, but I'm not sure it matters.